Source/JavaScriptCore/runtime/VM.h

```diff
@@ -1086,6 +1094,9 @@ class VM : public ThreadSafeRefCounted<VM>, public DoublyLinkedListNode<VM> {
     Ref<Waiter> m_syncWaiter;
 
     Vector<Function<void()>> m_didPopListeners;
+#if ENABLE(WEBASSEMBLY)
+    ThreadSafeWeakHashSet<Wasm::Instance> m_wasmInstances;
```

Comment:
This is WeakHashSet. So Wasm::Instance will be removed automatically when it dies.

Reply:
Just to confirm: there is a 1 to 1 correspondence between VM and WasmInstance, correct?

Reply:
No. WasmInstance -> VM. But no VM -> WasmInstance.

Reply:
The part I want to make sure of is that a `WasmInstance` can only be associated with 1 VM, correct?  It is never associated with more than 1 VM at the same time, correct?

Reply:
Never mind.  Yusuke pointed out offline that the commit msg already addressed this: "Wasm::Instance is strictly tied to one VM."